tracer/src/Datadog.Trace/Tagging/TagsList.cs

```diff
@@ -81,34 +88,41 @@ public override string ToString()
             return StringBuilderCache.GetStringAndRelease(sb);
         }
 
+        // Tags
+
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        protected static void WriteTag(ref byte[] bytes, ref int offset, string key, string value)
+        protected void WriteTag(ref byte[] bytes, ref int offset, string key, string value)
```

Comment:
I feel like this abstraction is all wrong now, and we should be passing in the _tagProcessors as a parameter to the `SerializeTo`/`WriteTag` methods. Setting the processors on the tag list, and then checking if there are any set feels wrong to me

Reply:
done in [63cb777](https://github.com/DataDog/dd-trace-dotnet/pull/2402/commits/63cb777d971a4c3f943740dc4d86f1e1a1edfe0b)